File: app/src/main/java/com/brankas/testapp/fragment/TransferDetailsFragment.java
package com.brankas.testapp.fragment;

import android.os.Bundle;

import com.brankas.testapp.R;
import com.brankas.testapp.TestApplication;
import com.brankas.testapp.customview.BoxedEditText;
import com.brankas.testapp.listener.ScreenListener;
import com.google.protobuf.Any;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class TransferDetailsFragment extends BaseFragment {

    public static int MAX_DESTINATION_ACCOUNT_ID = 36;
    public static String DESTINATION_ACCOUNT_ID = "destination_account_id";
    public static String MEMO = "memo";
    public static String AMOUNT = "amount";
    public static String REFERENCE_ID = "reference_id";

    @Override
    public int getLayoutId() {
        return R.layout.fragment_transfer_details;
    }

    @Override
    public void initDetails() {
        initDestinationAccountIdMaxLength();
    }

    @Override
    public void showError(String tag) {
        if(tag != null)
            ((BoxedEditText) parentView.findViewById(R.id.destinationAccountId)).showError(
                    getString(R.string.invalid_destination_account));
        else
            ((BoxedEditText) parentView.findViewById(R.id.destinationAccountId)).hideError();
    }

    @Override
    public void autoFill() {
        ((BoxedEditText) parentView.findViewById(R.id.destinationAccountId)).setText(
                TestApplication.getInstance().getDestinationAccountId());
        ((BoxedEditText) parentView.findViewById(R.id.memo)).setText("Bank Transfer");
        ((BoxedEditText) parentView.findViewById(R.id.amount)).setText("100");
        ((BoxedEditText) parentView.findViewById(R.id.referenceId)).setText("Sample Reference 1975");
    }

    @Override
    public HashMap<String, Object> getFieldsMap() {
        HashMap<String, Object> hashMap = new HashMap<>();
        hashMap.put(DESTINATION_ACCOUNT_ID, parentView.findViewById(R.id.destinationAccountId));
        hashMap.put(MEMO, parentView.findViewById(R.id.memo));
        hashMap.put(AMOUNT, parentView.findViewById(R.id.amount));
        hashMap.put(REFERENCE_ID, parentView.findViewById(R.id.referenceId));
        return hashMap;
    }

    @Override
    public List<String> getOptionalFields() {
        return new ArrayList<>();
    }

    @Override
    public int getPage() {
        return 1;
    }

    private void initDestinationAccountIdMaxLength() {
        ((BoxedEditText) parentView.findViewById(R.id.destinationAccountId))
                .updateMaxLength(MAX_DESTINATION_ACCOUNT_ID);
    }

    public void addAmountPrefix(String currency) {
        ((BoxedEditText) parentView.findViewById(R.id.amount)).showPrefix(currency);
    }

    public static TransferDetailsFragment newInstance(ScreenListener screenListener) {
        TransferDetailsFragment transferDetailsFragment = new TransferDetailsFragment();
        Bundle bundle = new Bundle();
        bundle.putParcelable(LISTENER, screenListener);
        transferDetailsFragment.setArguments(bundle);
        return transferDetailsFragment;
    }
}

File: app/src/main/java/com/brankas/testapp/fragment/TransferDetailsFragment.kt
package com.brankas.testapp.fragment

import com.brankas.testapp.R
import com.brankas.testapp.TestAppApplication
import com.brankas.testapp.`interface`.ScreenListener
import com.brankas.testapp.customview.BoxedEditText
import java.util.*

class TransferDetailsFragment: BaseFragment() {
    private lateinit var destinationAccountId: BoxedEditText
    private lateinit var memo: BoxedEditText
    private lateinit var amount: BoxedEditText
    private lateinit var referenceId: BoxedEditText

    override fun getLayoutId(): Int {
        return R.layout.fragment_transfer_details
    }

    override fun initDetails() {
        destinationAccountId = parentLayout.findViewById(R.id.destinationAccountId)
        memo = parentLayout.findViewById(R.id.memo)
        amount = parentLayout.findViewById(R.id.amount)
        referenceId = parentLayout.findViewById(R.id.referenceId)
        initDestinationAccountIdMaxLength()
    }

    override fun showError(tag: String?) {
        tag?.let {
            destinationAccountId.showError(getString(R.string.invalid_destination_account))
        } ?: run {
            destinationAccountId.hideError()
        }
    }

    override fun autoFill() {
        destinationAccountId.setText(TestAppApplication.instance.getDestinationAccountId())
        memo.setText("Bank Transfer")
        amount.setText("100")
        referenceId.setText(UUID.randomUUID().toString())
    }

    override fun getFieldsMap(): HashMap<String, Any> {
        val hashMap = hashMapOf<String, Any>()
        hashMap[DESTINATION_ACCOUNT_ID] = destinationAccountId
        hashMap[MEMO] = memo
        hashMap[AMOUNT] = amount
        hashMap[REFERENCE_ID] = referenceId
        return hashMap
    }

    override fun getOptionalFields(): List<String> {
        return emptyList()
    }

    override fun getPage(): Int {
        return 1
    }

    private fun initDestinationAccountIdMaxLength() {
        destinationAccountId.updateMaxLength(MAX_DESTINATION_ACCOUNT_ID)
    }

    fun addAmountPrefix(currency: String) {
        amount.showPrefix(currency)
    }

    companion object {
        const val MAX_DESTINATION_ACCOUNT_ID = 36
        const val DESTINATION_ACCOUNT_ID = "destination_account_id"
        const val MEMO = "memo"
        const val AMOUNT = "amount"
        const val REFERENCE_ID = "reference_id"

        fun newInstance(screenListener: ScreenListener): TransferDetailsFragment {
            return newInstance<TransferDetailsFragment>(screenListener)
        }
    }
}